server/src/internalClusterTest/java/org/opensearch/gateway/TransportNodesListGatewayStartedShardsBatchIT.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway;
+
+import org.opensearch.action.admin.cluster.shards.ClusterSearchShardsGroup;
+import org.opensearch.action.admin.cluster.shards.ClusterSearchShardsResponse;
+import org.opensearch.action.admin.cluster.state.ClusterStateRequest;
+import org.opensearch.action.admin.cluster.state.ClusterStateResponse;
+import org.opensearch.action.support.ActionTestUtils;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.index.Index;
+import org.opensearch.core.index.shard.ShardId;
+import org.opensearch.indices.store.ShardAttributes;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_NUMBER_OF_REPLICAS;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_NUMBER_OF_SHARDS;
+import static org.opensearch.gateway.AsyncShardFetchTestUtils.corruptShard;
+import static org.opensearch.gateway.AsyncShardFetchTestUtils.prepareRequestMap;
+
+public class TransportNodesListGatewayStartedShardsBatchIT extends OpenSearchIntegTestCase {
```

Comment:
These are functional tests for this transport action only. ```RecoveryFromGatewayIT``` class integ test for async fetch as well. Do you want to move these tests are and also test end to end functionality.

Reply:
As these test are only testing the functionality related to Transport that's why these tests were added to this class, moving these tests to `RecoveryFromGatewayIT` should not be a problem. But, as these transport classes are not getting used in this PR, end to end tests only make sense after changes in GatewayAlloctor are merged in #8746.

Reply:
Moved the tests to RecoveryFromGatewayIT